### ISSUE TITLE
XEP-0368: SRV records for XMPP over TLS failing

### ISSUE THREAD

**Issue by friesenkiwi** [iNPUTmice/ComplianceTester#71]
Hi,
I setup an own XMPP server (Prosody 0.9.10) at uberspace some time ago to establish a replacement for WhatsApp/Threema/Signal in my social environment, which went quite well so far in the Conversations/Android world. The ComplianceTester and IM Observatory both gave green light. Now that there is a OMEMO enabled iOS app, I wanted to extend it towards Chatsecure, but failed, due to an obscure error:
```
Domain: NSPOSIXErrorDomain
Code: 2
UserInfo: {
    NSLocalizedDescription = "No such file or directory";
    NSLocalizedFailureReason = "Error in connect() function";
}
```
I don't have any Apple device, so I have to check at friends. Since Conversations and Gajim are working without any flaw, I was quite surprised and couldn't find an explanation. So I tried the ComplianceTester again, finding it fails the latest test for the latest XEP. I didn't manage to find a meaningful error description (except for "FAILED"), or a complete explanation or example on why one would need it and how exactly to configure it.
So I tried to simply clone the old SRV records to new ones with an additional s:
```
_xmpp-client._tcp.im.pokerazor.de. 3600		IN SRV	10 0 64248 im.pokerazor.de.
_xmpps-client._tcp.im.pokerazor.de. 3600	IN SRV	10 0 64248 im.pokerazor.de.

_xmpp-server._tcp.im.pokerazor.de. 3600		IN SRV	10 0 65006 im.pokerazor.de.
_xmpps-server._tcp.im.pokerazor.de. 3600	IN SRV	10 0 65006 im.pokerazor.de.
```

But this doesn't seem to help the tester.

What did I do wrong? How to properly configure for XEP-0368 to make Chatsecure work?
Cheers already!

**Comment by iNPUTmice** [comment]
1) The result of that check has nothing to do with your ChatSecure issues
2) You can not just clone these records to pass the test

**Comment by friesenkiwi** [comment]
Thank you for looking into it! :-)
1. Yes, I now understand that these are unrelated, I will try to solve the problem over at ChatSecure in parallel
2. Still I would like the test to succeed. So what do I need to do? From what I now understand, I would need to configure the server (Prosody 0.9.10) to open two additional ports where it expects and accepts TLS only connections, without the need of STARTTLS before? I wasn't able to find such options for the prosody.cfg.lua file.

**Comment by ronnicek** [comment]
Hello @friesenkiwi :) did you solve it somehow? :)

**Comment by friesenkiwi** [comment]
1. No, unfortunately not, I did investigate a bit more but didn't open another bug report yet.
It seems, the problem is in ChatSecure's registration code. Connecting with an already existing account works without problems.
2. I did not really try to do it, because it seems unimportant to me. I think, the correct option within Prosody should be "legacy_ssl_ports" (see https://prosody.im/doc/configure)
